### PR TITLE
feat: Add support for Python in JSII and upgrade CDK version

### PR DESCRIPTION
**Pull request by LeonLuttenberger**
### Detail
- Upgraded CDK version to `v2.34.0`
- Added configuration for JSII to generate Python objects
- Changed module name to `aws-cdk-core`, to reflect the name of the Python package.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -98,3 +98,28 @@ jobs:
...
+          name: build-artifact
+          path: dist
+      - name: Prepare Repository
+        run: mv dist .repo
+      - name: Install Dependencies
+        run: cd .repo && yarn install --check-files --frozen-lockfile
+      - name: Create python artifact
+        run: cd .repo && npx projen package:python
+      - name: Collect python Artifact
+        run: mv .repo/dist dist
```
Comment (malachi-constant) [inline .github/workflows/build.yml:125]: Is this going to overwrite or be overwritten by the `package-js` step? Maybe related.. I noticed the build artifact in the PR checks is only typescript

```
> tree ~/Downloads/build-artifact -L 1
/Users/hansonlu/Downloads/build-artifact
├── API.md
├── CODE_OF_CONDUCT.md
├── LICENSE
├── NOTICE
├── README.md
├── coverage
├── lib
├── package.json
├── src
├── test
├── test-reports
├── tsconfig.dev.json
├── tsconfig.json
└── yarn.lock
```

File: .projen/deps.json [hunk 2]
```diff
@@ -97,7 +97,7 @@
     },
     {
       "name": "aws-cdk-lib",
-      "version": "^2.1.0",
+      "version": "^2.34.0",
```
Comment (malachi-constant) [inline .projen/deps.json:100]: Should we accelerate to `2.38.0` while we are at it? Appears to be [`cdk@latest`](https://docs.aws.amazon.com/cdk/api/versions.html)

File: .projen/tasks.json [hunk 3]
```diff
@@ -140,6 +143,15 @@
         }
       ]
     },
+    "package:python": {
+      "name": "package:python",
+      "description": "Create python language bindings",
+      "steps": [
+        {
+          "exec": "jsii-pacmak -v --target python"
```
Comment (malachi-constant) [inline .projen/tasks.json:151]: Magic 😂

**Review by malachi-constant — APPROVED** [review]
Looks good to me. I left a couple comments but nothing pressing.

**Review by kukushking — APPROVED** [review]
Nice! I would just go for it and enable other languages too.

File: .github/workflows/build.yml (hunk 1 above)
Comment (LeonLuttenberger) [reply .github/workflows/build.yml:125]: I don't think so. It generates two separate sets of artifacts. One with the Python bindings and one with the Typescript bindings.

**Pull request merged**